tests/unit/test_enums.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def test_compression_enums():
+    """Ensures that the compression enum has all the allowed types of
+    compression algorithms present.
+    """
+
+    from google.cloud.bigquery.enums import Compression
+
+    expected_comps_sorted = ["DEFLATE", "GZIP", "NONE", "SNAPPY", "ZSTD"]
+
+    result_comps_sorted = sorted(comp.value for comp in Compression)
```

Comment:
This is failing on Python 3.7 with `TypeError: 'type' object is not iterable`.

Compression isn't a true enum. It's an `object` with some handy class attributes. Now that we're Python 3.x-only, we should consider making these all [StrEnum](https://docs.python.org/3/library/enum.html#enum.StrEnum) classes.

Reply:
`StrEnum` was only introduced in Python 3.11, if memory serves, so it won't work here.

Trying a simple `enum.Enum` in a previous commit introduced errors but in other tests:

```
self = 

    def test_to_api_repr(self):
        from google.cloud.bigquery import job
    
        config = self._make_one()
        config.compression = job.Compression.SNAPPY
        config.destination_format = job.DestinationFormat.AVRO
        config.field_delimiter = "ignored for avro"
        config.print_header = False
        config._properties["extract"]["someNewField"] = "some-value"
        config.use_avro_logical_types = True
        resource = config.to_api_repr()
>       self.assertEqual(
            resource,
            {
                "extract": {
                    "compression": "SNAPPY",
                    "destinationFormat": "AVRO",
                    "fieldDelimiter": "ignored for avro",
                    "printHeader": False,
                    "someNewField": "some-value",
                    "useAvroLogicalTypes": True,
                }
            },
        )
E       AssertionError: {'ext[18 chars]on': , 'destinat[134 chars]rue}} != {'ext[18 chars]on': 'SNAPPY', 'destinationFormat': 'AVRO', 'f[112 chars]rue}}
E       - {'extract': {'compression': ,
E       ?                             ---------------------        -
E       
E       + {'extract': {'compression': 'SNAPPY',
E                      'destinationFormat': 'AVRO',
E                      'fieldDelimiter': 'ignored for avro',
E                      'printHeader': False,
E                      'someNewField': 'some-value',
E                      'useAvroLogicalTypes': True}}
```

Reply:
We definitely need string comparison to work.

Let's try this trick which should work for older Python verions:

```python
class MyEnum(str, enum.Enum):
    state1 = 'state1'
    state2 = 'state2'
```

https://stackoverflow.com/a/58608362/101923

That said, given this is only adding a constant, I'm fine without an explicit test for it if we want to keep the type as "object".